core/txpool/legacypool/legacypool_test.go

```diff
@@ -198,8 +200,8 @@ func validatePoolInternals(pool *LegacyPool) error {
 		if nonce := pool.pendingNonces.get(addr); nonce != last+1 {
 			return fmt.Errorf("pending nonce mismatch: have %v, want %v", nonce, last+1)
 		}
-		if txs.totalcost.Cmp(common.Big0) < 0 {
-			return fmt.Errorf("totalcost went negative: %v", txs.totalcost)
+		if txs.totalcost.Cmp(max) > 0 {
```

Comment:
This does not make sense. A `uint256.Int` is limited between `0` and `max` (`ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff`). It is inherently not possible for it to go above `max`. 

Making a change like this, where we swap out a type is not just a matter of making it compile and done. You need to actually spend some time considering the consequences of the changes. Some checks may no longer be required, and in some places maybe we need to add new checks.

Reply:
My bad, I should've been more careful. In my opinion, we can remove this check since it's not possible to check for an overflow here.